cdap-app-fabric-tests/src/test/java/io/cdap/cdap/internal/app/runtime/batch/DynamicPartitionerWithAvroTest.java

```diff
@@ -184,6 +184,7 @@ public void apply() throws IOException {
 
     if (!expectedStatus) {
       // if we expect the program to fail, no need to check the output data for expected results
+      deleteApp(app.getApplicationId());
```

Comment:
why do you need changes in this class?

Reply:
We have to delete the app, since we allow only deployment of the same app if the config differs from the latest version (which it doesn't in most cases)